File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
      map<int,int> mp;
      multiset<pair<int,int>> freq_order;//sorting freq_wise
      for(int i=0;i<nums.size();i++){
        if(freq_order.find(make_pair(mp[nums[i]],-nums[i]))!=freq_order.end()){
            freq_order.erase(freq_order.find(make_pair(mp[nums[i]],-nums[i])));
        }
        mp[nums[i]]++;
        freq_order.insert(make_pair(mp[nums[i]],-nums[i]));
      
      }
      vector<int> ans;
      int count=k;
      while(count--){
        ans.push_back(-(freq_order.rbegin()->second));
        freq_order.erase(prev(freq_order.end()));
      }
      return ans;
    }
};

File: Data Structures & Algorithms/subarray-sums-divisible-by-k/submission-0.java
class Solution {
    public int subarraysDivByK(int[] nums, int k) {
        HashMap<Integer,Integer> mp=new HashMap<>();
        int[] prefix=new int[nums.length];
        mp.put(0,1);
        prefix[0]=nums[0];
        for(int i=1;i<nums.length;i++){
            prefix[i]=nums[i]+prefix[i-1];
        }
        int count=0;
        for(int i=0;i<prefix.length;i++){
            int ans=prefix[i]%k;
            if(ans<0){
                ans+=k;
            }
            if(mp.containsKey(ans)){
                count+=mp.get(ans);
            }
            mp.put(ans,mp.getOrDefault(ans,0)+1);

        }
        return count;
    }
}